internal/locate/region_request.go

```diff
@@ -1275,15 +1268,11 @@ func (s *replicaSelector) canFallback2Follower() bool {
 	if s == nil || s.state == nil {
 		return false
 	}
-	state, ok := s.state.(*accessFollower)
-	if !ok {
-		return false
-	}
-	if !state.isStaleRead {
+	_, ok := s.state.(*accessKnownLeader)
+	if ok {
 		return false
 	}
-	// can fallback to follower only when the leader is exhausted.
-	return state.lastIdx == state.leaderIdx && state.IsLeaderExhausted(s.replicas[state.leaderIdx])
+	return true
```

Comment:
This may skip too much backoff, be careful, need more discuss.

Reply:
Better to leave this change to another PR, the logic is quite different.